target/linux/ramips/dts/mt7621_adslr_g7.dts

```diff
@@ -105,13 +105,41 @@
 	};
 };
 
-&ethernet {
+&gmac0 {
 	mtd-mac-address = <&factory 0xe00c>;
 };
 
+&switch0 {
+	ports {
+		port@0 {
+			status = "okay";
+			label = "lan1";
+		};
+		port@1 {
+			status = "okay";
+			label = "lan2";
+		};
+		port@2 {
+			status = "okay";
+			label = "lan3";
+		};
+		port@3 {
+			status = "okay";
+			label = "lan4";
+		};
+		port@4 {
+			status = "okay";
+			label = "wan";
+			mtd-mac-address = <&factory 0xe00c>;
+			mtd-mac-address-increment = <1>;
+		};
+	};
+};
+
+
 &state_default {
 	gpio {
-		ralink,group = "i2c", "wdt";
-		ralink,function = "gpio";
+		groups = "i2c", "uart3", "wdt";
```

Comment:
uart3 added here and other dts should be removed as well. It should only be added to devices actually using gpio8 and gpio9 as separated pcie reset pins.
When needed, it should be added as a separated pinctrl node like existing pcie_pins node in mt7621.dtsi instead of writing it in state_default.

Reply:
Do you mean overlay pcie pins adding "uart3" group in specific board dts file? Like this?

```
+&pcie_pins {
+       groups = "pcie", "uart3";
+       function = "gpio";
+};
+
 &pcie {
        pinctrl-names = "default";
        pinctrl-0 = <&pcie_pins>;
@@ -124,7 +129,7 @@ &pcie {
 &pinctrl {
        state_default: pinctrl0 {
                default_gpio: gpio {
-                       groups = "wdt", "rgmii2", "uart3";
+                       groups = "wdt", "rgmii2";
                        function = "gpio";
                };
```

Reply:
ADSLR G7 actually uses uart3 (GPIO 7 and 8)
I will remove the rest of them

Reply:
Your dts change is incorrect.
Here's the original pcie_pins node in mt7621.dtsi:

```dts
		pcie_pins: pcie {
			pcie {
				groups = "pcie";
				function = "gpio";
			};
		};
```

With your override, it becomes:

```dts
		pcie_pins: pcie {
			groups = "pcie", "uart3";
			function = "gpio";
			pcie {
				groups = "pcie";
				function = "gpio";
			};
		};
```

The node in dts should either be:

```dts
	&pcie_pins {
		pcie {
			groups = "pcie", "uart3";
			function = "gpio";
		};
	};
```

or:

```dts
	&pcie_pins {
		uart3_gpio {
			groups = "uart3";
			function = "gpio";
		};
	};
```

Reply:
Yes, you are totally right. Thanks for explanation. But:

```
--- a/drivers/staging/mt7621-dts/gbpc1.dts
+++ b/drivers/staging/mt7621-dts/gbpc1.dts
@@ -111,6 +111,13 @@ &cpuclock {
                        clock-frequency = <900000000>;
 };
 
+&pcie_pins {
+       pcie {
+               groups = "pcie", "uart3";
+               function = "gpio";
+       };
+};
+
 &pcie {
        pinctrl-names = "default";
        pinctrl-0 = <&pcie_pins>;
@@ -124,7 +131,7 @@ &pcie {
 &pinctrl {
        state_default: pinctrl0 {
                default_gpio: gpio {
-                       groups = "wdt", "rgmii2", "uart3";
+                       groups = "wdt", "rgmii2";
                        function = "gpio";
                };
        };
```

I got the following error:

```
[   16.285795] rt2880-pinmux pinctrl: pcie is already enabled
[   16.296729] mt7621-pci 1e140000.pcie: Error applying setting, reverse things back
[   16.311634] rt2880-pinmux pinctrl: not freeing pin 19 (io19) as part of deactivating group pcie - it is already used for some other setting
[   16.336706] mt7621-pci 1e140000.pcie: Parsing DT failed
```

I don't really understand why this change affects to #gpio19 also and the kernel is complaining about it.